### PR TITLE
ROX-19483: Add default email subject, body and config details to emailed reports

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

Same as title

## Checklist
- [ ] Investigated and inspected CI test results
- [X] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

- Unit tests

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by sachaudh** [comment]
I added @zhenpesky to take a look at the screenshots as well.

I think the text before the details (CVE Severity, etc.) should have a bit more spacing. It's too close. Also, the text seems to be center-aligned for the details. Can you make it align to the top?

Screenshot 2023-09-14 at 10 27 06 AM

**Comment by sachaudh** [comment]
@zhenpesky @charmik-redhat @ksurabhi91 @md2119 How would you feel if we kept the report parameters section simple like this? 

Screenshot 2023-09-14 at 11 23 26 AM

I think it's easier to parse this visually. I felt like I had to jump around with my eyes a lot when looking at the report parameters values. Since this is also a new feature, might as well keep it simple for now, and adjust as customers give input. Maybe they'd be fine with something simple like this

**Comment by zhenpesky** [comment]
> I think it's easier to parse this visually. I felt like I had to jump around with my eyes a lot when looking at the report parameters values. Since this is also a new feature, let's keep it simple for now, and adjust as customers give input. Maybe they'd be okay with something simple like this (or might not have a strong opinion anyway).
> 
> Also, backend won't need to do any fancy styling for this.

@sachaudh Yes. LGTM. I think it's better to do it in the way you described especially since those details are in a modal view where the viewport is smaller.

**Review by ksurabhi91 — COMMENTED** [review]
Add a comment to pass empty string for v1 config

**Comment by zhenpesky** [comment]
@sachaudh @charmik-redhat out of curiosity, the image you have in the email client view (stackrox email).  How do we ensure the format layout is consistent across different email platforms? does it always maintain 2 rows in headers with details in 3 column layout no matter which email service they use to open the email?

**Comment by md2119** [comment]
> @zhenpesky @charmik-redhat @ksurabhi91 @md2119 @vjwilson How would you feel if we kept the report parameters section simple like this?
> 
> Screenshot 2023-09-14 at 11 23 26 AM
> 
> I think it's easier to parse this visually. I felt like I had to jump around with my eyes a lot when looking at the report parameters values. Since this is also a new feature, let's keep it simple for now, and adjust as customers give input. Maybe they'd be okay with something simple like this (or might not have a strong opinion anyway).
> 
> Also, backend won't need to do any fancy styling for this.

I prefer a simplified format. The markdowns and icons pose limitations in other notifiers, which seems unnecessary. 

One change request: There is no restriction on report name and collection name length. Therefore, it is possible that it may be trimmed from the subject. When that happens, users will have lost that information. My recommendation is to include the Report Configuration Name and Collection Name in the email body along with the currently shown parameter list.

**Comment by ksurabhi91** [comment]
> @zhenpesky @charmik-redhat @ksurabhi91 @md2119 @vjwilson How would you feel if we kept the report parameters section simple like this?
> 
> Screenshot 2023-09-14 at 11 23 26 AM
> 
> I think it's easier to parse this visually. I felt like I had to jump around with my eyes a lot when looking at the report parameters values. Since this is also a new feature, let's keep it simple for now, and adjust as customers give input. Maybe they'd be okay with something simple like this (or might not have a strong opinion anyway).
> 
> Also, backend won't need to do any fancy styling for this.

I like this simpler format much better. And will definitely save backend from writing a lot of html

**Comment by vjwilson** [comment]
> @zhenpesky @charmik-redhat @ksurabhi91 @md2119 @vjwilson How would you feel if we kept the report parameters section simple like this?
> 
> Screenshot 2023-09-14 at 11 23 26 AM
> 
> I think it's easier to parse this visually. I felt like I had to jump around with my eyes a lot when looking at the report parameters values. Since this is also a new feature, let's keep it simple for now, and adjust as customers give input. Maybe they'd be okay with something simple like this (or might not have a strong opinion anyway).
> 
> Also, backend won't need to do any fancy styling for this.

I'm fine with this format.

File: central/reports/scheduler/v2/reportgenerator/email_formatter.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+		for i, valI := range values {
+			writer.WriteString(convertValueToFriendlyText(valI))
+			if i < (len(values) - 1) {
+				writer.WriteString(", ")
+			}
+		}
+		writer.WriteString("</span>")
+	}
+
+	writer.WriteString("</div>")
```
Comment (md2119) [inline central/reports/scheduler/v2/reportgenerator/email_formatter.go:163]: Just a simple multi-line "key": "value" pair structure would remove the markup requirement altogether and would work with all notifiers.

**Review by md2119 — APPROVED** [review]

**Review by sachaudh — APPROVED** [review]

File: central/reports/scheduler/v2/reportgenerator/email_formatter.go (hunk 1 above)
Comment (charmik-redhat) [reply central/reports/scheduler/v2/reportgenerator/email_formatter.go:163]: I think the email does look better with some markup added. But I also agree that it should be usable by other notifiers as well. In a followup PR, I would try to add a text based table (no markup) and that pattern can be used for other notifiers as well.

**Comment by charmik-redhat** [comment]
> One change request: There is no restriction on report name and collection name length. Therefore, it is possible that it may be trimmed from the subject. When that happens, users will have lost that information. My recommendation is to include the Report Configuration Name and Collection Name in the email body along with the currently shown parameter list.

Agree. The Collection name is already included in the Report scope section. I have added the Report Configuration name to the config details.
cc @sachaudh @zhenpesky

**Pull request merged**